### PR TITLE
5105 missing select mulitple answers that used search expression

### PR DESCRIPTION
**Pull request by ivermac**

### CODE REVIEW

File: src/hatti/ona/forms.cljs [hunk 1]
```diff
@@ -215,9 +215,20 @@
                                           first)
                               formatted (get-label option language)]
                           (or formatted answer))
-    (select-all? field) (let [names (set (string/split answer #" "))]
-                          (->> (:children field)
-                               (filter #(contains? names (field-key %)))
+    (select-all? field) (let [names (set (string/split answer #" "))
+                              appearance-value (-> field :control :appearance)
+                              uses-search-expression?
+                              (and (string? appearance-value)
```
Comment (pld) [inline src/hatti/ona/forms.cljs]: let's move this to a helper function, since that's the purpose it's service here

**Comment by ivermac** [comment]
Added a helper function to check if field has search expression

File: src/hatti/ona/forms.cljs [hunk 2]
```diff
@@ -215,9 +221,17 @@
                                           first)
                               formatted (get-label option language)]
                           (or formatted answer))
-    (select-all? field) (let [names (set (string/split answer #" "))]
-                          (->> (:children field)
-                               (filter #(contains? names (field-key %)))
+    (select-all? field) (let [names (set (string/split answer #" "))
+                              appearance-value (-> field :control :appearance)
+                              has-search-expression? (uses-search-expression?
```
Comment (pld) [inline src/hatti/ona/forms.cljs]: This doesn't need a var binding as far as I can tell, since it's only used once

**Comment by ivermac** [comment]
I have refactored the code by removing the var binding

**Pull request merged**